src/tonic/api/points_internal_api.rs

```diff
@@ -605,17 +609,19 @@ impl PointsInternal for PointsInternalService {
                     }
                 },
             };
-            let mut response = result.into_inner();
+            let response = result.into_inner();
 
-            if let Some(usage) = response.usage.take() {
+            if let Some(usage) = response.clone().usage.unwrap_or_default().hardware.take() {
```

Comment:
I'd prefer something like this to prevent unnecessary cloning.
```suggestion
            let mut response = result.into_inner();
            let mut usage = response.usage.take();

            if let Some(usage) = usage.as_mut().and_then(|i| i.hardware.take()) {
                total_usage.add(usage);
            }

```